service/src/integrationTest/kotlin/fi/espoo/evaka/invoicing/controller/FinanceDecisionCitizenIntegrationTest.kt

```diff
@@ -0,0 +1,326 @@
+import fi.espoo.evaka.invoicing.data.upsertValueDecisions
+import fi.espoo.evaka.invoicing.domain.ChildWithDateOfBirth
+import fi.espoo.evaka.invoicing.domain.DecisionIncome
+import fi.espoo.evaka.invoicing.domain.FeeAlterationWithEffect
+import fi.espoo.evaka.invoicing.domain.FeeDecision
+import fi.espoo.evaka.invoicing.domain.FeeDecisionChild
+import fi.espoo.evaka.invoicing.domain.FeeDecisionStatus
+import fi.espoo.evaka.invoicing.domain.FeeDecisionThresholds
+import fi.espoo.evaka.invoicing.domain.FeeDecisionType
+import fi.espoo.evaka.invoicing.domain.FinanceDecisionType
+import fi.espoo.evaka.invoicing.domain.VoucherValueDecision
+import fi.espoo.evaka.invoicing.domain.VoucherValueDecisionPlacement
+import fi.espoo.evaka.invoicing.domain.VoucherValueDecisionServiceNeed
+import fi.espoo.evaka.invoicing.domain.VoucherValueDecisionStatus
+import fi.espoo.evaka.invoicing.domain.VoucherValueDecisionType
+import fi.espoo.evaka.invoicing.testFeeThresholds
+import fi.espoo.evaka.placement.PlacementType
+import fi.espoo.evaka.shared.ChildId
+import fi.espoo.evaka.shared.DaycareId
+import fi.espoo.evaka.shared.FeeDecisionId
+import fi.espoo.evaka.shared.PersonId
+import fi.espoo.evaka.shared.VoucherValueDecisionId
+import fi.espoo.evaka.shared.auth.AuthenticatedUser
+import fi.espoo.evaka.shared.auth.CitizenAuthLevel
+import fi.espoo.evaka.shared.dev.DevPerson
+import fi.espoo.evaka.shared.dev.DevPersonType
+import fi.espoo.evaka.shared.dev.insert
+import fi.espoo.evaka.shared.domain.DateRange
+import fi.espoo.evaka.shared.domain.FiniteDateRange
+import fi.espoo.evaka.shared.domain.Forbidden
+import fi.espoo.evaka.shared.domain.HelsinkiDateTime
+import fi.espoo.evaka.shared.domain.MockEvakaClock
+import fi.espoo.evaka.snDaycareFullDay35
+import fi.espoo.evaka.testChild_1
+import fi.espoo.evaka.testChild_2
+import fi.espoo.evaka.testDaycare
+import fi.espoo.evaka.toFeeDecisionServiceNeed
+import fi.espoo.evaka.toValueDecisionServiceNeed
+import java.math.BigDecimal
+import java.time.LocalDate
+import java.time.LocalTime
+import java.util.UUID
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.beans.factory.annotation.Autowired
+
+class FinanceDecisionCitizenIntegrationTest : FullApplicationTest(resetDbBeforeEach = true) {
+
+    @Autowired private lateinit var applicationControllerCitizen: ApplicationControllerCitizen
+
+    private val clock =
+        MockEvakaClock(HelsinkiDateTime.of(LocalDate.of(2022, 10, 23), LocalTime.of(21, 0)))
+
+    private lateinit var headOfFamily: DevPerson
+    private lateinit var partner: DevPerson
+
+    private lateinit var feeDecisions: List<FinanceDecisionCitizenInfo>
+    private lateinit var voucherValueDecisions: List<FinanceDecisionCitizenInfo>
+
+    @BeforeEach
+    fun beforeEach() {
+        db.transaction { tx ->
+            tx.insertGeneralTestFixtures()
```

Comment:
It would be better to avoid `insertGeneralTestFixtures` in new code, but you don't have to change it if you don't want to. `insertGeneralTestFixtures` cannot be deprecated because non-suppressed deprecations are compile errors 🙄

Reply:
Replaced mass insert for fixtures by more selective inserts.